src/proto/macro.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto2";
+
+package mixxx.track.io;
+
+option optimize_for = LITE_RUNTIME;
+
+message Macro {
+  message Action {
+    optional uint32 type = 1;
+    optional uint64 source_frame = 2;
+    optional uint64 target_frame = 3;
```

Comment:
Hmm, maybe these should be required. Not sure. My understanding of protobuf is that this does not affect the serialized output, just the deserialization code. So we should probably make these required and "downgrade" them to optional only if the becomes necessary.

Reply:
@daschuer @Swiftb0y Do you know Protobuf well enough to answer this question with certainty?

Reply:
Not with certainty, but until now my rule of thump was that everything should be optional just in case. In protobuf 3 the concept of required fields has been removed altogether: 
https://stackoverflow.com/questions/31801257/why-required-and-optional-is-removed-in-protocol-buffers-3